module-utils/time/time/time_conversion.cpp

```diff
@@ -207,15 +207,44 @@ namespace utils::time
             return Timestamp::day();
         }
         if (isCurrentYear()) {
-            auto localeFormat = timeSettings.isDateFormatDDMM() ? Locale::TimeFormat::FormatLocaleDate_DD_MM
-                                                                : Locale::TimeFormat::FormatLocaleDate_MM_DD;
+            const auto localeFormat = timeSettings.isDateFormatDDMM() ? Locale::TimeFormat::FormatLocaleDate_DD_MM
+                                                                      : Locale::TimeFormat::FormatLocaleDate_MM_DD;
             return Timestamp::str(Locale::format(localeFormat));
         }
-        auto localeFormat = timeSettings.isDateFormatDDMM() ? Locale::TimeFormat::FormatLocaleDate_DD_MM_YYYY
-                                                            : Locale::TimeFormat::FormatLocaleDate_MM_DD_YYYY;
+        const auto localeFormat = timeSettings.isDateFormatDDMM() ? Locale::TimeFormat::FormatLocaleDate_DD_MM_YYYY
+                                                                  : Locale::TimeFormat::FormatLocaleDate_MM_DD_YYYY;
         return Timestamp::str(Locale::format(localeFormat));
     }
 
+    UTF8 DateAndTime::str(std::string format) const
+    {
+        if (format.compare("") != 0) {
```

Comment:
Maybe better just `empty()`? Or is there a special purpose to do it this way here?

Reply:
done